Feature/#6 공통 header footer

Handsol opened the pull request:
### 관련 이슈
- close #6

### 작업 요약
- Header, Footer 부분 꾸미기 구현 완료

### 리뷰 요구 사항
- 리뷰 예상 시간 `10분`

### 미리보기
- Header 
![image](https://github.com/user-attachments/assets/6d2d558a-efc5-4a2a-92c9-806cc042f54a)
- Footer
![image](https://github.com/user-attachments/assets/a220d0a9-1e6c-4d8d-ba3f-2d617a3b2e84)

Handsol committed:
🎉 add : 로고이미지, 팀 깃허브, 노션 이미지 추가

#6

Handsol committed:
🎉 add : 공통 Header 추가

#6

Handsol committed:
🎉 add : 공통 Footer 추가

#6

llddang commented on layout/footer/footer.css:
```diff
@@ -0,0 +1,45 @@
+footer {
+    width: 100vw;
+    background-color: #484747;
+    position: fixed;
+    bottom: 0;
```
개인적인 생각으로 footer의 경우 `position: fixed`가 필요없다고 생각합니다.
`position: fixed`는 말 그대로 화면의 어느 부분에 고정하겠다라는 의미인데요.

footer의 경우 무조건 한 화면에 보여야하는 것이 아닌, 그저 main 내용 밑에 있어야 하는 부분이므로,
나중에 `<main> ... </main> {footer 내용}` 이렇게 작업하는게 좋을 것 같습니다!

제가 말을 잘 못해서 혹시 이해 안 되는 부분 있으면 comment 달아주세요 :)

Handsol replied:
그러면 main이나 다른 화면들에 크기가 확정되어있으니 코드를 Merge 할 경우 footer가 화면 최하단에 붙는다는 말씀이신거죠? 그러면 굳이 고정할 필요가 없겠군요..!! 하나 배워갑니다!

llddang commented on layout/footer/footer.css:
```diff
@@ -0,0 +1,45 @@
+
+.footerTop {
+    display: flex;
+    flex-direction: row;
```
flex의 기본 방향이 row이기 때문에 없어도 될 것 같습니다!

Handsol replied:
오 이런 알겠습니다! column이나 다른 방향을 지정할 경우에만 써줘야겠군요

jiyxxn replied:
row 아니면 column 밖에 없기 때문에 column이 필요한 경우에만 작성해주시면 됩니다 👍

llddang commented on layout/footer/footer.css:
```diff
@@ -0,0 +1,45 @@
+
+.footerLogo {
+    width: 250px;
+    filter: grayscale(100%);
```
오홍 흑백 전환할 때 이런 속성이 있군요!
배워갑니다!!

Handsol replied:
^ㅁ^

llddang commented on layout/footer/footer.css:
```diff
@@ -0,0 +1,45 @@
+    flex-direction: row;
+    width: 1260px;
+    color: #939393;
+}
```
맨 마지막에 지금 빈 줄이 없어서 깃헙 에러가 나고 있습니다.

마지막에 빈 줄을 넣어야 하는 이유에 대한 [블로그](https://velog.io/@jangws/EOL%EC%9D%84-%EB%84%A3%EC%96%B4%EC%95%BC-%ED%95%98%EB%8A%94-%EC%9D%B4%EC%9C%A0%EC%99%80-git%EC%97%90%EC%84%9C-CRLF-EOL-%EC%B0%A8%EC%9D%B4%EB%A1%9C-%EC%9D%B8%ED%95%9C-%EB%AC%B8%EC%A0%9C-%ED%95%B4%EA%B2%B0-%EB%B0%A9%EB%B2%95)입니다!
참고해주세용

vscode에서는 확장 프로그램으로 prettier 설치하고 `format on save: true` & `formatter` 설정하면 저장할 때마다 자동으로 잡아줄 거예요!

Handsol replied:
마지막 줄에 엔터 한번 더 쳐서 빈 줄을 만들어야하는군요! 확인하고 전체 EOL 추가했습니다!

llddang commented on layout/header/header.css:
```diff
@@ -0,0 +1,59 @@
+}
+
+.header {
+    width: 100vW;
```
사소한 것이지만... 단위 `vw`로 맞춰주실 수 있을까요?

Handsol replied:
넵! `vW`로 돼있던 부분 `vw`로 수정하겠습니다

llddang reviewed (changes requested):
수고하셨습니다!
지금 모든 파일에서 EOL 에러가 나고 있어서요.
VScode 확장프로그램으로 Prettier 설치하고 다시 저장한 다음에 올려주시면 좋을 것 같아요!
![image](https://github.com/user-attachments/assets/6fbbd9f6-728c-4531-8a6a-e98a3d883163)

Handsol committed:
T✨ update  : Header에 기능 추가

해당하는 페이지로 넘어갈때 네비게이션 바에 언더라인 나타나는 기능 구현
Custom Web Component 추가
#6

Handsol committed:
✨ update :  footer에 js 기능 추가

Custom Web Component 기능 추가하여 index.html과 연결
width : 100vW로 적었던 부분 단위 수정 -> width : 100vw
모든 파일에 마지막 빈줄 추가 (기존에 없었음)
position : fixed 제거
flex 방향으로 지정했던 row 제거
#6

jiyxxn commented on layout/header/header.css:
```diff
@@ -51,9 +51,16 @@ a {
+    cursor: pointer;
+}
+
+.underline {
+    color: #5588C6;
+    text-decoration: underline;
```
`underline` 이라는 속성을 클래스화해서 적용하는 방법 아주 좋습니다! ☺
다만, 클래스명을 `underline`이라고 지어주셨기 때문에 해당 클래스 안에는 실제로 '밑줄' 기능만 포함하는 것이 좋습니다.
다른 협업자가 밑줄 속성을 사용하기 위해 `.underline`을 사용했는데 의도치 않은 텍스트 색상 변경이 함께 발생하게 되면 당황하겠죠?!
만약 gnb 내 메뉴가 활성화 되었을 때 해당 속성이 필요한 거라면, 클래스명을 `underline`이라고 짓기 보다는 `active` 로 지어보시는 것을  추천합니다! 그러면 '활성화된'이라는 의미이기 때문에 underline과 color 변경이 동시에 발생해도 의미적으로 충돌하지 않을 것 같네요!

클래스명을 속성명과 동일하게 작성할 경우에는 딱 그 속성만! 넣어주는 게 좋다는 점 기억해주세요 👍

Handsol replied:
오 active로 지으면 훨씬 더 직관적이고 헷갈리지 않겠네요! 조언 감사합니다!

jiyxxn commented on layout/header/header.html:
```diff
@@ -9,21 +9,34 @@
         <div class="headerTop">
             <h1>
                 <img src="../../public/logo.png" class="headerLogo">
+                <audio controls loop src="../../public/짱구.mp3"></audio>
```
저희의 배경음악이 들어간 구간이군요 !
다만 추후에 파일을 생성하고 작업하실 때는 파일명을 공백이 없는 영문으로 작성하는 것을 권장합니다!
비록 컴퓨터가 한글을 인식할 수 있는 경우도 있지만, 실제로 한글은 컴퓨터에서 유니코드라는 **숫자 코드**로 처리됩니다. 이 때문에 한글을 포함한 파일명을 사용하면, 때로는 파일을 제대로 찾지 못하는 문제가 발생할 수 있습니다. 

따라서, 코드 작업을 좀 더 원활하게 진행하려면 앞으로 파일명을 영문(공백없이)으로 작성하는 것을 추천드려요!

Handsol replied:
코딩할땐 파일명을 공백없는 영문으로 메모!

jiyxxn replied:
공백 대신 언더스코어(`_`)를 사용하거나 하이픈(`-`) 또는 카멜케이스로 작성하시는 방법이 있답니다 🎅

jiyxxn commented on layout/header/header.html:
```diff
@@ -9,21 +9,34 @@
+                <li class="navTeam"><a href="../../member/home.html">내배캠 방범대</a></li>
+                <li class="navMember"><a href="../../member/member.html">방범대원 소개</a></li>
+                <li class="navGuestbook"><a href="../../guestbook/guestbook.html">방범 일지</a></li>
             </ul>
```
이 부분만 하이라이팅하긴 했지만 전체적으로 구조를 꼼꼼하게 짜주신 것 같네요 !!
영역이 깔끔하게 나누어져있고 영역마다 명확한 이름으로 클래스명을 지어주셔서 구조 확인이 쉬웠습니다.
고생하셨어요!

Handsol replied:
꼼꼼하게 리뷰해주셔서 감사합니다!!! ^ㅁ^

Handsol committed:
✨ update : 헤Header, Footer를 index.html에 적용

- header과 footer이 화면 제자리에 위치할 수 있도록 추가함
#6

Handsol committed:
✨ update : 배경 오디오 컨트롤러 위치 변경

:컨트롤러의 기존 위치인 헤더 로고 옆에서 네비게이션 우측으로 이동
#6

Handsol committed:
�� del : header.html 파일 삭제

필요없는 html 파일 삭제
#6

Handsol committed:
� del : footer.html 파일 삭제

필요없는 html 파일 삭제
#6

Handsol committed:
✨ update : b홈 메인화면 바디 사이즈 수정

height 사이즈 수정 및 min-height 800px 내용 추가
#6

Handsol committed:
� add : BGM 추가

재생될 짱구 관련 BGM 추가
header.js에 내용 반영 완료
#6

Handsol committed:
♻️ refactor : index.html 리팩토링

#6

Handsol committed:
�del : 사용하지 않는 이미지 삭제

#6

llddang commented on layout/header/header.css:
```diff
@@ -0,0 +1,74 @@
+
+.headerLogo {
+    width: 250px;
+
```
빈줄은 지워주세용 :)

llddang commented on layout/header/header.css:
```diff
@@ -0,0 +1,74 @@
+    right: 5%;
+    top: 50%;
+    transform: translateY(-50%);
+}
```
오잉 이유는 모르겠지만 formatter가 적용이 안된 것 같아요!

https://hyeon9mak.github.io/github-no-newline-at-a-end-of-file/
한번 읽어봐주세용!

지금은 개행없이 끝나도 문제가 되지 않습니다.
하지만 예전에는 개행에 따라 문제가 발생할 수 있어서 파일의 끝에 개행을 넣어주었습니다.
그 행위가 지금은 관습으로 남았어요!
(블로그.. 이전에 알려줬던 것 보다 내용이 괜찮은 것 같아 남깁니다!)

llddang commented on layout/header/header.js:
```diff
@@ -0,0 +1,46 @@
+    `;
+  }
+  afterLogic() {
+    //이한솔
```
ㅋㅋㅋㅋㅋ한솔님이 하셨군요!!

llddang commented on layout/header/header.js:
```diff
@@ -0,0 +1,46 @@
+
+    menuItems.forEach((item) => {
+      const menuPage = item.getAttribute("href").split("/").pop();
+      console.log(menuPage);
```
머지할 때는 console.log 삭제해주세용 :)

llddang reviewed (approved):
수고하셨습니다!

리펙토링 약간만 하면 될 것 같아요!

Handsol committed:
✨ update :  li 전체로 지정된 css 스타일 수정

li 대상으로 cursor: pointer가 지정되어있었던 부분 확인하여 li를 nav li로 대상 재지정
#6

Handsol committed:
✨ update : hjs파일에 있던 console.log 내용 삭제

css파일에 개행 추가
hederLogo 스타일에 있던 빈 줄 삭제
js 파일에 주석으로 적어뒀던 이름 삭제
확인을 위해 넣어뒀던 console.log 내용 삭제
#6

Sumin-Lee12 reviewed (approved):
코드랑 코멘드 보면서 많이 배워갑니다!!
